### PR TITLE
feat(helm): deploy postgres as part of infra chart

### PR DESCRIPTION
**Pull request by mxyng**
## Summary

Set up and configure postgres in helm. Needs #2568 to have most of the value since that PR removes the other requirement for a infra-server PVC

### TODO

- [x] update infra configmap to use postgres

### Tested

- [x] Setting out of band database
- [x] Setting out of band password with `dbPasswordSecret`
- [x] Users/groups migration
- [x] Destinations migration
- [x] Providers migration
- [x] Access tokens migration
- [ ] Organizations migration
- [x] Rollback

### CODE REVIEW

File: helm/charts/infra/templates/postgres/_helpers.tpl [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+*/}}
+{{- define "postgres.envFrom" -}}
+{{- concat .Values.postgres.envFrom .Values.global.envFrom | uniq | toYaml }}
+{{- end }}
+
+{{/*
+Infer whether postgres should be deployed based on postgres.enabled and connector.config.postgres.
+*/}}
+{{- define "postgres.enabled" -}}
+{{- and .Values.postgres.enabled .Values.server.enabled }}
```
[the remark this answers is not in the document]
Comment (mxyng) [reply helm/charts/infra/templates/postgres/_helpers.tpl:115]: Definitely. This is one of the todos I haven't gotten around to doing

File: helm/charts/infra/templates/postgres/_helpers.tpl [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+postgres 'env' values. Merges global and local values.
+*/}}
+{{- define "postgres.env" -}}
+{{- $env := concat .Values.postgres.env .Values.global.env }}
+{{- $env = append $env (dict "name" "POSTGRES_DB" "value" .Values.postgres.dbName) }}
+{{- $env = append $env (dict "name" "POSTGRES_USER" "value" .Values.postgres.dbUsername) }}
+{{- if .Values.postgres.dbPasswordSecret -}}
+{{- $env = append $env (dict "name" "POSTGRES_PASSWORD" "valueFrom" (dict "secretKeyRef" (dict "name" .Values.postgres.dbPasswordSecret "key" "password"))) }}
+{{- else }}
+{{- $env = append $env (dict "name" "POSTGRES_PASSWORD" "valueFrom" (dict "secretKeyRef" (dict "name" (include "postgres.fullname" .) "key" "password"))) }}
```
Comment (BruceMacD) [inline helm/charts/infra/templates/postgres/_helpers.tpl:107]: Could this result in an empty password being set?

File: helm/charts/infra/templates/postgres/statefulset.yaml [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+          resources:
+{{- toYaml .Values.postgres.resources | nindent 12 }}
+{{- if .Values.server.persistence.enabled }}
+      initContainers:
+        - name: migration
+          image: alpine:latest
+          command: [sh, -c]
+          args:
+            - |
+              apk add --no-cache sqlite
```
Comment (BruceMacD) [inline helm/charts/infra/templates/postgres/statefulset.yaml:74]: Do we still need sqlite?

**Review by BruceMacD — APPROVED** [review]
Looks good, just called out some sqlite stuff that I believe has changed since this PR was opened.

File: helm/charts/infra/templates/postgres/statefulset.yaml (hunk 3 above)
Comment (mxyng) [reply helm/charts/infra/templates/postgres/statefulset.yaml:74]: Yes, this is for the migration

**Pull request merged**